src/docs/patterns/forms.mdx

```diff
@@ -1,36 +1,70 @@
+import dismissKeyboardErrorTextImage from '../../app/assets/design-patterns/form-validation/dismiss-keyboard-error-text.gif';
 
 <!-- KEYWORDS: design pattern form verification validation error -->
 
 # Form Validation Patterns
 
-<ImageGrid images={[Image1, Image2, Image3]} />
+<ImageGrid images={[SummaryImage]} regularWidth />
 
-Forms are an integral part of many applications, particularly during the user registration phase. One of the most important parts of working with forms is properly validating the information that users enter into them. In general, PX Blue follows the form behaviors explained in detail by the Material Design.
+Forms are an integral part of many applications, particularly during the [user registration](/patterns/login) phase. One of the most important parts of working with forms is properly validating the information that users enter into them. In general, PX Blue follows the form behaviors explained in detail by Material Design.
 
-<MaterialDesignDescription title={'Text Fields'} url={'https://material.io/components/text-fields/'} />
-<DemoCard 
-    repository={'form-validation'} 
-    angular 
-    react={'https://codesandbox.io/s/github/pxblue/react-design-patterns/tree/master?file=/src/pages/form-validation/index.tsx&initialpath=form-validation'} 
-/>
+<React.Fragment>
+    <MaterialDesignDescription title={'Text Fields'} url={'https://material.io/components/text-fields/'} />
+    <DemoCard
+        repository={'form-validation'}
+        angular
+        react={
+            'https://codesandbox.io/s/github/pxblue/react-design-patterns/tree/master?file=/src/pages/form-validation/index.tsx&initialpath=form-validation'
+        }
+    />
+    <Divider />
+</React.Fragment>
+
+# Behaviors
 
 ## Optional vs. Required Fields
 
 When using a mixture of optional and required fields in a form, the inputs should be labeled appropriately.
 
 If most of the fields are required, you should call out the fields that are optional. These fields should be labeled with an appropriate descriptor, followed by "optional" in parentheses.
 
-Conversely, if most of your fields are optional, you should only call out the required fields. These should be identified by an asterisk (\*) after the field label. There should be a descriptor nearby that indicates that a \* means the field is required.
+Conversely, if most of your fields are optional, you should only call out the required fields. These should be identified by an asterisk (\*) after the field label. Optionally, you can add a descriptor nearby that indicates that a \* means the field is required.
+
+## Form Length
+
+The form should not be too lengthy -- no more than the screen height on the mobile device, and no more than half of the desktop screen. Consider breaking long forms into subsections or multiple pages using tabs and [steppers](/patterns/steppers).
```

Comment:
Subsections are mostly so that people will have a chance to breathe in the middle of a long form, and make it easier to skim read/navigate.

Reply:
Instead of making it two rules, would it be easier to just say something like:
"Keep your forms short where possible. If your forms are larger than the height of your target device, consider breaking them into sections or multiple pages to improve the user experience"